R/zchunk_L261.Cstorage.R

```diff
@@ -56,204 +70,309 @@ module_energy_L261.Cstorage_DISABLED <- function(command, ...) {
+    A61.globaltech_shrwt %>%
+      write_to_all_regions(LEVEL2_DATA_NAMES[["Tech"]],
+                           GCAM_region_names = GCAM_region_names) %>%
+      select(region, supplysector, subsector, stub.technology = technology) ->
+      L261.StubTech_C # This is a final output table.
+
+    # Energy inputs and coefficients of global technologies for carbon storage
+    # A61.globaltech_coef reports carbon storage global technology coefficients
+    A61.globaltech_coef %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>% # Convert to long form
+      mutate(year = as.integer(year)) %>% # Year needs to be integer or numeric to interpolate
+      # Expand table to include all model base and future years
+      complete(year = c(year, MODEL_YEARS), nesting(supplysector, subsector, technology, minicam.energy.input)) %>%
+      # Extrapolate to fill out values for all years
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(coefficient = approx_fun(year, value, rule = 2)) %>%
+      filter(year %in% MODEL_YEARS) %>% # This will drop 1971
+      # Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      select(sector.name = supplysector, subsector.name = subsector, technology, year, minicam.energy.input, coefficient) ->
+      L261.GlobalTechCoef_C # This is a final output table.
+
+    # Costs of global technologies
+    # A61.globaltech_cost reports carbon storage offshore storage cost (1975$/tCO2)
+    A61.globaltech_cost %>%
+      gather(year, value, matches(YEAR_PATTERN)) %>% # Convert to long form
+      mutate(year = as.integer(year)) %>% # Year needs to be integer or numeric to interpolate
+      # Expand table to include all model base and future years
+      complete(year = c(year, MODEL_YEARS), nesting(supplysector, subsector, technology, minicam.non.energy.input)) %>%
+      # Extrapolate to fill out values for all years
+      # Rule 2 is used so years outside of min-max range are assigned values from closest data, as opposed to NAs
+      mutate(input.cost = approx_fun(year, value, rule = 2)) %>%
+      filter(year %in% MODEL_YEARS) %>% # This will drop 1971
+      # Assign the columns "sector.name" and "subsector.name", consistent with the location info of a global technology
+      select(sector.name = supplysector, subsector.name = subsector, technology, year, minicam.non.energy.input, input.cost) ->
+      L261.GlobalTechCost_C # This is a final output table.
+
+    # High costs of global technologies for carbon storage -- this prices out CCS
+    L261.GlobalTechCost_C %>%
+      mutate(subsector.name = "onshore carbon-storage",
+             technology = "onshore carbon-storage") %>%
+      bind_rows(L261.GlobalTechCost_C) %>%
+      mutate(input.cost = 10000) -> # 1975$/tCO2
```

Comment:
Added comment. Anything more specific and I'll have to open an issue. Let me know whatever is best.